blocks/bricks/recurrent.py

```diff
@@ -180,12 +182,21 @@ def recurrent_apply(brick, application, application_call,
             for name, state in states_given.items():
                 states_given[name] = tensor.unbroadcast(state,
                                                         *range(state.ndim))
+            if config.scan_strict:
+                # Find all shared variables
+                shared_vars = list(Selector(brick).get_params().values())
+                # Just make `scan_function` discard them before calling
+                # `application_function`
+                strict = {'strict': True}
```

Comment:
Yes, I do. I still haven't had a chance to check if this PR gets any improvement.